src/amo/components/AddonReviewCard/index.js

```diff
@@ -481,7 +481,8 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
           </a>
         ) : null}
 
-        {flaggable && review ? (
+        {/* Do not render the "flag" button for replies made by the current user */}
+        {flaggable && review && siteUser && siteUser.id !== review.userId ? (
```

Comment:
If the user is signed out, the *Flag* link should always be visible. Otherwise, if someone wants to flag a review, they might not know that they have to sign in first.

This is what it looks like (there is actually a style regression that I will fix right now! https://github.com/mozilla/addons-frontend/issues/6656)

<img width="1372" alt="screenshot 2018-10-16 11 45 43" src="https://user-images.githubusercontent.com/55398/47033150-b41de800-d139-11e8-99f6-87c15ac0f15d.png">